.github/workflows/continuous-integration.yml

```diff
@@ -7,6 +7,9 @@ on:
 env:
   AWS_SWIFT_SDK_USE_LOCAL_DEPS: 1
 
+permissions:
+  id-token: write
+
```

Comment:
are you sure we can't get rid of these extra permissions in CI?  IIRC this is needed to get AWS credentials through OpenID which we don't do in the CI workflow.

Reply:
On the linux side I'm using `aws-actions/configure-aws-credentials@v3` still. Would rather I try to change this one to fake credentials?

Reply:
Yes, let's not get real credentials where fake ones will do